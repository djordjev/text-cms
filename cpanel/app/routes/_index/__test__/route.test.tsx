import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import Route from '../route';

test('Route', async () => {
  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => <Route />,
        loader() {
          return [
            {
              type: 'heading',
              children: [{ text: 'Heading' }],
              level: 2
            },
            {
              type: 'paragraph',
              children: [{ text: 'Paragraph' }]
            }
          ];
        }
      }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(await screen.findByText('Heading')).toBeInTheDocument();
  expect(screen.getByText('Paragraph')).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
