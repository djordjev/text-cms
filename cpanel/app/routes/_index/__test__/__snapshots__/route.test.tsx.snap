// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Route 1`] = `
NodeList [
  <div
    class="u-p-4x"
  >
    <div>
      <h2
        class="u-text-3xl"
      >
        <span
          class=""
        >
          Heading
        </span>
      </h2>
      <p
        class=""
      >
        <span
          class=""
        >
          Paragraph
        </span>
      </p>
    </div>
  </div>,
]
`;
